app/models/ems_event.rb

```diff
@@ -33,6 +33,27 @@ def self.group_names_and_levels
     end
   end
 
+  def self.in_group(group)
+    if group == DEFAULT_GROUP_NAME
+      all
+    else
+      by_literal, by_regex = event_types_by_group
+
+      scope = where(:event_type => by_literal[group])
+      by_regex[group].each do |regex|
+        scope = scope.or(where("event_type ~* ?", regex))
```

Comment:
we will be generating so many `OR` clauses that it feels like we have no chance of avoiding a scan. But maybe we'll be lucky and scan the `event_type` index.

Reply:
I've removed this for now as it was just a PoC of how to do query-level filtering by group and group_level.